Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1997,6 +1997,15 @@ RefPtr<API::Navigation> WebPageProxy::loadRequest(WebCore::ResourceRequest&& req
 #endif
 
     loadRequestWithNavigationShared(protectedLegacyMainFrameProcess(), m_webPageID, navigation, WTFMove(request), shouldOpenExternalURLsPolicy, isPerformingHTTPFallback, userData, ShouldTreatAsContinuingLoad::No, isNavigatingToAppBoundDomain(), std::nullopt, std::nullopt);
+
+#if ENABLE(CONTENT_EXTENSIONS)
+    if (preferences().iFrameResourceMonitoringEnabled()) {
+        Ref processPool = m_legacyMainFrameProcess->processPool();
+        if (RefPtr ruleList = processPool->cachedResourceMonitorRuleList())
+            protectedLegacyMainFrameProcess()->send(Messages::WebProcess::SetResourceMonitorContentRuleList(ruleList->data()), 0);
```

Comment:
Seems like this should be only sent if the process hasn't already gotten the list in the past.

Reply:
I'll move back the initialization to webprocessproxy to remember the status. And from `loadRequestWithNavigationShared` to send a request to process proxy to load that.